File: AdventuraClick.Model/Reservation.cs
namespace AdventuraClick.Model
{
    public class Reservation
    {
        public int ReservationId { get; set; }
        public string? Status { get; set; }
        public DateTime? Date { get; set; } = null!;
        public int NumberOfPassengers { get; set; }
        public int? TravelId { get; set; }
        public virtual Travel? Travel { get; set; }
        public int? UserId { get; set; }
        public virtual User? User { get; set; }
        public int? PaymentId { get; set; }
        public virtual Payment? Payment { get; set; }
        public int? TravelInformationId { get; set; }
        public virtual TravelInformation? TravelInformation { get; set; }
        public List<AdditionalService> AdditionalServices { get; set; }
        public virtual ICollection<AdditionalServiceReservation> AdditionalServicesReservations { get; set; } = new List<AdditionalServiceReservation>();

    }
}

File: adventuraclick_mobile/lib/model/reservation.dart
import 'package:adventuraclick_mobile/model/additional_services.dart';
import 'package:adventuraclick_mobile/model/travel.dart';
import 'package:json_annotation/json_annotation.dart';
part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int? reservationId;
  int? userId;
  int? travelId;
  int? paymentId;
  String? status;
  DateTime? date;
  Travel? travel;

  Reservation();

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  /// Connect the generated [_$RatingToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}